detection_rules/rule.py

```diff
@@ -971,11 +992,11 @@ def latest_version(self) -> Optional[int]:
     @property
     def autobumped_version(self) -> Optional[int]:
         """Retrieve the current version of the rule, accounting for automatic increments."""
-        version = self.latest_version
+        version = self.data.version or self.latest_version
         if version is None:
             return 1
 
-        return version + 1 if self.is_dirty else version
+        return version + 1 if self.is_dirty and RULES_CONFIG.version_strategy == "auto" else version
```

Comment:
Here's an approach to distinguish which versions are used based on a field set in the `RULES_CONFIG`

Reply:
I like this :+1:  adds the potential/hook for users to define their own versioning strategy as well.

Reply:
the 3 strategies we have discussed being supportable are:
1. version lock
2. explicitly set within `rule.version`
3. defer to kibana (never set)

2 and 3 must ignore all version lock functionality 

We should add it to the default config commented out and I would change it to a bool variable like `RULES_CONFIG.ignore_version_lock`

IINM, there are likely several other places that are dependent on version lock use that will need to be decoupled

Reply:
also need to decide what to do if they _dont_ `ignore_version_lock` but _do_ explicitly set versions (error, ignore, warn, etc.)

Reply:
I updated the summary so far with what is implemented. 


- General (bypass_version_lock = False)
    - Default
    - Versions from Kibana or the TOML file are ignored
    - Version lock file usage is permitted

- General (bypass_version_lock = True)
    - Must be explicitly set in the config
    - Versions from Kibana or the TOML file are used
    - Version lock file usage is not permitted

- Tactical Warning Messages
    - Rule import to TOML file will skip version and revision fields when supplied (rule_prompt & import_rules_into_repo) if `bypass_version_lock = False`. No warning message is issued.
    - Rule version lock will not be updated or used if `bypass_version_lock = True` when building a release package (build_release). A warning message is issued.
    - If versions are in the TOML file, and `bypass_version_lock = False`, the versions in the TOML file will not be used (autobumped_version). A warning message is issued. 
    - If `bypass_version_lock = False`, when autobumping the version, it will check the version lock file and increment if `is_dirty` (autobumped_version), otherwise just use the version supplied. No warning message is issued.

I'll update it again if we make changes before merging.